app/lib/forem_stats_drivers/datadog_driver.rb

```diff
@@ -0,0 +1,37 @@
+module ForemStatsDrivers
+  class DatadogDriver
+    attr_reader :driver
+
+    def initialize
+      Datadog.configure do |c|
+        c.tracer env: Rails.env
+        c.tracer enabled: ENV["DD_API_KEY"].present?
+        c.tracer partial_flush: true
+        c.tracer priority_sampling: true
+        c.use :elasticsearch
+        c.use :sidekiq
+        c.use :redis
+        c.use :rails
+        c.use :http
+      end
+
+      @driver = Datadog::Statsd.new
+    end
+
+    def increment(*args)
```

Comment:
**suggestion (non-blocking):** Same comment re delegation as above.  Makes me wonder if we want to share behavior e.g via concern:

```ruby
module ActsAsForemStatsDriver
  extend ActiveSupport::Concern

  class_methods do
    def setup_driver(&setup)
      define_method(:initialize) do
        @driver = setup.call
      end
    end
  end

  included do
    delegate :increment, :time, :gauge, to: :@driver
    alias_method :count, :increment
  end
end
```

This would greatly simplify adding more drivers while still allowing to override or alias methods, etc:

```ruby
module ForemStatsDrivers
  class DatadogDriver
    setup_driver do
      Datadog.configure do |c|
        ...
      end
      Datadog::Statsd.new
    end
 end
end
```

This is not really tested or anything, but the throwaway example I wrote in my Rails console seems to work fine:

```ruby
class ExampleService; def increment(x); x + 1; end; end

class ExampleServiceDriver
  include ActsAsForemStatsDriver
  setup_driver { ExampleService.new }
end

ExampleServiceDriver.new.increment(1)
#=> 2
ExampleServiceDriver.new.public_methods(false)
#=> [:count, :gauge, :increment, :time]
```

Reply:
:eyes:  Let me do some quick testing on my end, too, as I have the New Relic driver stuff also close-to-ready so that I can deliver it as a separate PR after this foundational one is good. But, my initial impression is that I think that this would be perfectly fine.  I'll follow up soon.